Submodule libraries/libwebsockets contains modified content
diff --git a/libraries/libwebsockets/lib/core/context.c b/libraries/libwebsockets/lib/core/context.c
index 64028b0f..0076eb62 100644
--- a/libraries/libwebsockets/lib/core/context.c
+++ b/libraries/libwebsockets/lib/core/context.c
@@ -1761,7 +1761,7 @@ lws_system_cpd_start_defer(struct lws_context *cx, lws_usec_t defer_us)
 			 lws_system_deferred_cb, defer_us);
 }
 
-#if (defined(LWS_WITH_SYS_STATE) && defined(LWS_WITH_SYS_SMD)) || !defined(LWS_WITH_NO_LOGS)
+#if (defined(LWS_WITH_SYS_STATE) && defined(LWS_WITH_SYS_SMD)) || (_LWS_ENABLED_LOGS & LLL_INFO)
 static const char *cname[] = { "Unknown", "OK", "Captive", "No internet" };
 #endif
 
@@ -1771,9 +1771,7 @@ lws_system_cpd_set(struct lws_context *cx, lws_cpd_result_t result)
 	if (cx->captive_portal_detect != LWS_CPD_UNKNOWN)
 		return;
 
-#if !defined(LWS_WITH_NO_LOGS)
 	lwsl_cx_info(cx, "setting CPD result %s", cname[result]);
-#endif
 
 	cx->captive_portal_detect = (uint8_t)result;
 
